### PR TITLE
Implement ADR-107: rename proto package version suffixes to v1beta* pattern

### PR DESCRIPTION
**Pull request by mzabaluev**
Closes: #1504

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Comment by sergio-mena** [comment]
Is this ready for review?

File: api/cometbft/blocksync/types.go [hunk 1]
```diff
@@ -2,19 +2,23 @@
 package blocksync
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/blocksync/v1"
-	v2 "github.com/cometbft/cometbft/api/cometbft/blocksync/v2"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/blocksync/v1beta1"
+	v1beta2 "github.com/cometbft/cometbft/api/cometbft/blocksync/v1beta2"
```
Comment (sergio-mena) [inline api/cometbft/blocksync/types.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/blocksync/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/blocksync/v1beta2"
```

File: api/cometbft/consensus/types.go [hunk 2]
```diff
@@ -2,33 +2,33 @@
 package consensus
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/consensus/v1"
-	v2 "github.com/cometbft/cometbft/api/cometbft/consensus/v2"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/consensus/v1beta1"
+	v1beta2 "github.com/cometbft/cometbft/api/cometbft/consensus/v1beta2"
```
Comment (sergio-mena) [inline api/cometbft/consensus/types.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/consensus/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/consensus/v1beta2"
```

File: api/cometbft/consensus/wal.go [hunk 3]
```diff
@@ -2,18 +2,22 @@
 package consensus
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/consensus/v1"
-	v2 "github.com/cometbft/cometbft/api/cometbft/consensus/v2"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/consensus/v1beta1"
+	v1beta2 "github.com/cometbft/cometbft/api/cometbft/consensus/v1beta2"
```
Comment (sergio-mena) [inline api/cometbft/consensus/wal.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/consensus/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/consensus/v1beta2"
```

File: api/cometbft/privval/types.go [hunk 4]
```diff
@@ -2,26 +2,30 @@
 package privval
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/privval/v1"
-	v2 "github.com/cometbft/cometbft/api/cometbft/privval/v2"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/privval/v1beta1"
+	v1beta2 "github.com/cometbft/cometbft/api/cometbft/privval/v1beta2"
```
Comment (sergio-mena) [inline api/cometbft/privval/types.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/privval/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/privval/v1beta2"
```

File: api/cometbft/state/types.go [hunk 5]
```diff
@@ -1,15 +1,17 @@
 package state
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/state/v1"
-	v3 "github.com/cometbft/cometbft/api/cometbft/state/v3"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/state/v1beta1"
+	v1beta3 "github.com/cometbft/cometbft/api/cometbft/state/v1beta3"
```
Comment (sergio-mena) [inline api/cometbft/state/types.go:5]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/state/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/state/v1beta3"
```

File: api/cometbft/types/canonical.go [hunk 6]
```diff
@@ -1,12 +1,14 @@
 package types
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/types/v1"
-	v3 "github.com/cometbft/cometbft/api/cometbft/types/v3"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
+	v1beta3 "github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```
Comment (sergio-mena) [inline api/cometbft/types/canonical.go:5]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```

File: api/cometbft/types/events.go [hunk 7]
```diff
@@ -1,7 +1,7 @@
 package types
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/types/v1"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
```
Comment (sergio-mena) [inline api/cometbft/types/events.go:4]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
```

File: api/cometbft/types/evidence.go [hunk 8]
```diff
@@ -2,14 +2,18 @@
 package types
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/types/v1"
-	v3 "github.com/cometbft/cometbft/api/cometbft/types/v3"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
+	v1beta3 "github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```
Comment (sergio-mena) [inline api/cometbft/types/evidence.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```

File: api/cometbft/types/params.go [hunk 9]
```diff
@@ -1,15 +1,17 @@
 package types
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/types/v1"
-	v2 "github.com/cometbft/cometbft/api/cometbft/types/v2"
-	v3 "github.com/cometbft/cometbft/api/cometbft/types/v3"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
+	v1beta2 "github.com/cometbft/cometbft/api/cometbft/types/v1beta2"
+	v1beta3 "github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```
Comment (sergio-mena) [inline api/cometbft/types/params.go:6]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta2"
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta3"
```

File: api/cometbft/types/validator.go [hunk 10]
```diff
@@ -1,16 +1,18 @@
 package types
 
 import (
-	v1 "github.com/cometbft/cometbft/api/cometbft/types/v1"
+	v1beta1 "github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
```
Comment (sergio-mena) [inline api/cometbft/types/validator.go:4]:
```suggestion
	"github.com/cometbft/cometbft/api/cometbft/types/v1beta1"
```

File: p2p/netaddress.go [hunk 11]
```diff
@@ -346,34 +347,35 @@ func (na *NetAddress) ReachabilityTo(o *NetAddress) int {
...
+// address prior to the .onion (i.e. the key hash) base32 into a ten
+// byte number. It then stores the first 6 bytes of the address as
+// 0xfd, 0x87, 0xd8, 0x7e, 0xeb, 0x43.
+//
+// This is the same range used by OnionCat, which is part part of the
+// RFC4193 unique local IPv6 range.
+//
+// In summary the format is:
+// { magic 6 bytes, 10 bytes base32 decode of key hash }
+var onionCatNet = ipNet("fd87:d87e:eb43::", 48, 128)
```
Comment (sergio-mena) [inline p2p/netaddress.go:377]: We could leave this inside the parenthesis above

File: types/block_test.go [hunk 12]
```diff
@@ -213,9 +213,11 @@ func makeBlockID(hash []byte, partSetSize uint32, partSetHash []byte) BlockID {
 var nilBytes []byte
 
 // This follows RFC-6962, i.e. `echo -n ” | sha256sum`
-var emptyBytes = []byte{0xe3, 0xb0, 0xc4, 0x42, 0x98, 0xfc, 0x1c, 0x14, 0x9a, 0xfb, 0xf4, 0xc8,
+var emptyBytes = []byte{
```
[the remark this answers is not in the document]
Comment (mzabaluev) [reply types/block_test.go]: This breaks the code: it's a global, I think the short form is not allowed there.

File: p2p/netaddress.go (hunk 11 above)
Comment (mzabaluev) [reply p2p/netaddress.go]: https://github.com/cometbft/cometbft/pull/1510/commits/aef1472bfd84b2c2f85a2eaa621677e6cc4fc216

Comment (sergio-mena) [reply p2p/netaddress.go]: Better not fight opinionated formatters :-)

File: types/block_test.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (sergio-mena) [reply types/block_test.go:216]: Oops! Didn't notice we're in global scope here. Please ignore the noise

**Comment by sergio-mena** [comment]
Ready to go! 🚀

**Comment by sergio-mena** [comment]
... don't bother fixing the linting (#1531)

**Pull request merged**